pyanaconda/dbus/typing.py

```diff
@@ -149,24 +149,30 @@ def _get_basic_type(type_hint):
     @staticmethod
     def _is_container_type(type_hint):
         """Is it a container type?"""
-        # Try to get the "base" type of the container type.
+        # Try to get the "origin" of the hint.
         origin = getattr(type_hint, "__origin__", None)
-        return origin in DBusType._container_type_mapping
+        if origin:
+            # Return true if the "origin" is a subclass of a container type
+            # see https://bugzilla.redhat.com/show_bug.cgi?id=1598574
+            return any(issubclass(origin, contype) for contype in DBusType._container_type_mapping)
```

Comment:
Please, create a new method `_get_container_base_type`, that will get a type hint and return a base type or None. Then you can simplify the `_is_container_type` and `_get_container_type` methods.